tdesign/src/Layout/MarketingNotice/MarketingNotice.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from "react";
+import { styled } from "@mui/material/styles";
+import type { muiTheme } from "@splitgraph/tdesign/src/themes/muiTheme";
+import { Button } from "@mui/material";
+
+interface CallToActionOpts {
+  href: string;
+}
+
+interface MarketingNoticeProps {
+  cta?: React.PropsWithChildren<CallToActionOpts>;
+}
+
+const MarketingNoticeContainer = styled("div")(
+  ({ theme }: { theme: ReturnType<typeof muiTheme> }) => ({
+    paddingTop: "14px",
+    paddingBottom: "14px",
+    paddingLeft: `calc(5px + ${theme.constants.leftMarginLogoAligned})`,
+    paddingRight: theme.constants.rightMarginNavAligned,
+    background: theme.constants.pinkBackgroundGradient,
+    display: "flex",
+    alignItems: "center",
+  })
+);
+
+const CenteredChildrenContainer = styled("div")({
+  color: "heavy.main",
+  flexGrow: 1,
+  display: "flex",
+  justifyContent: "center",
+});
+
+const RightAlignedCTAContainer = styled("div")({
+  alignSelf: "flex-end",
+});
+
+const CTAButton = styled(Button)(
+  ({ theme }: { theme: ReturnType<typeof muiTheme> }) => ({
```

Comment:
I didn't check, but do we need to specify the type of `theme` here :worried: ? I was hoping the declaration merging would work the same way it works in the parent repo, where we don't need to pass the theme type manually (e.g. see `AddExternalPluginCredentialPageContent` and the use of `styled` there)

No need to change it in this PR, I'm just asking in general

Reply:
Nevermind, I checked - looks like it is necessary :disappointed:

Reply:
Yes. Are you sure that declaration merging works in the parent repo? weird if it works there and not here. Probably also indicative of something happening twice...

Reply:
Yup, I am sure, in the corresponding parent repo MR I linked to a use of `styled` that used the declaration-merged `Theme` type. I will try to take a look if I'll be modifying some code in this repo